pydoclint/utils/generic.py

```diff
@@ -76,6 +76,19 @@ def detectMethodType(node: ast.FunctionDef) -> MethodType:
     return MethodType.INSTANCE_METHOD
 
 
+def checkIsAbstractMethod(node: ast.FunctionDef) -> bool:
+    """Check whether `node` is an abstract method"""
+    if len(node.decorator_list) == 0:
+        return False
+
+    for decorator in node.decorator_list:
+        if isinstance(decorator, ast.Name):
+            if decorator.id == 'abstractmethod':
```

Comment:
You could also use `__isabstractmethod__` here (https://docs.python.org/3/library/abc.html#abc.abstractmethod), which might be slightly more robust in case another decorator has been used (e..g `@abstractproperty`, even though it's now deprecated)

Reply:
You actually touched on an interesting implementation detail of pydoclint.

Pydoclint performs static syntax analysis using Python's `ast` module.  "Static" means that the Python code is not actually run.

For example, when `ast` parses this:

```python
@abstractmethod
def something(self, arg):
    pass
```

It does not actually know that it is an abstract method.  It only knows that the method `something` has a decorator called "abstractmethod".

If you attempt to do this:

```python
hello = abstractmethod

@hello
def something(self, arg):
    pass
```

`ast` will only know that the method `something` has a decorator called "hello".

By now you must have realized that there is a type of potential edge cases: if people do `hello = staticmethod`, `world = abstractmethod`, etc., pydoclint will get confused and throw false positive violations.

But in reality, very few people do that.  (Would you?)

To be able to understand that "hello" means "staticmethod" in the code, we'd need to use Python's official `inspect` module.  I chose not to, because using `inspect` to parse source code is much slower than using `ast`.  (This is not difficult to understand: with `inspect`, we need to first get the source code running.)

Darglint uses `inspect` (see code examples [here](https://github.com/search?q=repo%3Aterrencepreilly%2Fdarglint+inspect.is&type=code) and [here](https://github.com/search?q=repo%3Aterrencepreilly%2Fdarglint+inspect.getmembers&type=code)).  I suspect this is why darglint is very slow.

What I plan to do is to summarize what I said above into a disclaimer section in pydoclint's documentation, so that users know what pydoclint is **not** designed to handle.